api-controllers/ServiceFabrikApiController.js

```diff
@@ -433,33 +369,39 @@ class ServiceFabrikApiController extends FabrikBaseController {
   abortLastBackup(req, res) {
     req.manager.verifyFeatureSupport('backup');
     const backupStartedAt = new Date();
-    return eventmesh.apiServerClient.getLastOperation({
-      resourceId: req.params.instance_id,
-      operationName: CONST.OPERATION_TYPE.BACKUP,
-      operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
-    }).then(backupGuid => {
-      return eventmesh.apiServerClient.getOperationState({
+    return eventmesh
+      .apiServerClient.getLastOperation({
+        resourceId: req.params.instance_id,
         operationName: CONST.OPERATION_TYPE.BACKUP,
         operationType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BACKUP,
-        operationId: backupGuid,
-      }).then(state => {
-        // abort only if the state is in progress
-        if (state === CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS) {
-          return eventmesh.apiServerClient.updateOperationState({
+      })
+      .then(backupGuid => {
```

Comment:
The method invoked method is `getLastOperation` and its response is expected to be a `backupguid`? rather it would be better if it returned an `operation` object something like the below, which is generic:
```
{
   operation: backup
   operation_params: {
        backup_guid: xxxx
   }
}
```
In the above you can skip `operation_parms` and directly put those op specific params one level up.  
Or if the intention is to always return backup_guid then name the method accordingly.

Reply:
This is like a foreign key for last backup of the deployment. We are getting the last backup operation guid of the deployment `req.params.instance_id`.
The above details are then queried using the backup_guid